Add theano.tensor.nnet.relu function

f0k opened the pull request:
As discussed in #2698, this adds a `theano.tensor.nnet.relu` function providing an optimized formulation of the linear rectifier and leaky linear rectifier at least for GPUs, with the potential to be extended with a different formulation for CPUs or even a ReLU Op in future.

/edit: I'm not sure about the docs, how can I test them?

nouiz commented on theano/tensor/nnet/tests/test_nnet.py:
```diff
@@ -1394,5 +1395,29 @@ def test_stabilize_log_softmax():
+
+def test_relu():
+    x = matrix('x')
+    rng = numpy.random.RandomState([2015, 6, 18])
```
Use this seed instead (this allow to change it with a Theano flags)

theano.tests.unittest_tools.fetch_seed()

f0k replied:
Okay. What about the test directly above? Shall I change that as well or leave it as is?

nouiz commented:
can you import it in the file `theano/tensor/nnet/__init__.py` that way, if we move it people won't need to udpate there code. (and it is easier to use it)

otherwise, it is pretty good. thanks.

f0k commented:
> can you import it in the file `theano/tensor/nnet/__init__.py`

That file has `from nnet import *` already, and `nnet.py` does not have an `__all__` statement. So my function is already available as `theano.tensor.nnet.relu` (along with `theano.tensor.nnet.numpy`, `theano.tensor.nnet.DisconnectedType` and a lot of other cruft, but that's outside the scope of this PR).

I've changed the random seed for the test as you asked for, and corrected some markup for the documentation.

nouiz commented:
thanks. I'll merge when travis pass.

f0k commented:
Travis is happy.
